Improve RBAC documentation (#11254)
kubernetes/website: content/en/docs/reference/access-authn-authz/rbac.md

  apiGroup: rbac.authorization.k8s.io
```

> The `Role` or `ClusterRole` a binding object refers to may not be modified.

Reviewer: ```suggestion
You cannot modify the `Role` or `ClusterRole` that a binding object refers to.
```
How about this wording?

Author: went with "You cannot modify which `Role` or `ClusterRole` a binding object refers to." instead

## Command-line Utilities

### `kubectl create role`

Creates a `Role` object defining permissions within a single namespace. Examples:

* Create a `Role` named "pod-reader" that allows user to perform "get", "watch" and "list" on pods:

    ```
    kubectl create role pod-reader --verb=get --verb=list --verb=watch --resource=pods
    ```

* Create a `Role` named "pod-reader" with resourceNames specified:

    ```
    kubectl create role pod-reader --verb=get --resource=pods --resource-name=readablepod --resource-name=anotherpod
    ```

* Create a `Role` named "foo" with apiGroups specified:

    ```
    kubectl create role foo --verb=get,list,watch --resource=replicasets.apps
    ```

* Create a `Role` named "foo" with subresource permissions:

    ```

>     kubectl create role foo --verb=get,list,watch --resource=pods,pods/status

Reviewer 2: Want to add an example that uses `--resource-name`?

Author: added
